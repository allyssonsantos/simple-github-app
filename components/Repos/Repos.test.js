import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

const getReposWithContext = context => {
  jest.doMock('../../providers/AppProvider', () => {
    return {
      AppConsumer: props => props.children(context),
    };
  });

  return require('./index').default;
};

describe('Form component', () => {
  const context = {
    openRepo: jest.fn(),
    userInfo: { login: 'test' },
    repos: [
      {
        id: 1,
        name: 'test',
        description: 'test description',
        stargazers_count: 1,
        updated_at: '2019-01-14T04:08:31Z',
      },
    ],
  };

  describe('Snapshots', () => {
    it('Should match snapshot', () => {
      const Repo = getReposWithContext(context);
      const component = renderer.create(<Repo />);
      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('Context functions', () => {
    it('Should call openRepo from context', () => {
      const Repo = getReposWithContext(context);
      const wrapper = mount(<Repo />);

      expect(context.openRepo).toBeCalledTimes(0);

      wrapper.find('RepoCard').simulate('click');

      expect(context.openRepo).toBeCalledTimes(1);
    });
  });
});
