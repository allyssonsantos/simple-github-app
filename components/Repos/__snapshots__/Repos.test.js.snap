// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component Snapshots Should match snapshot 1`] = `
.c1 {
  color: #000;
  font-size: inherit;
  margin: 0;
  margin-bottom: 10px;
}

.c4 {
  color: #000;
  font-size: 16px;
  margin: 0;
  margin-bottom: 10px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  background-color: #fff;
  border-radius: 3px;
  box-shadow: 0px 0px 1px #000;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  min-height: 150px;
  padding: 10px;
}

.c0 {
  border-top: 1px solid #ccc;
  margin-top: 20px;
  padding-top: 12px;
}

.c2 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(250px,1fr));
  grid-gap: 10px;
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    Repositories
  </h3>
  <main
    className="c2"
  >
    <article
      className="c3"
      onClick={[Function]}
    >
      <header>
        <h2
          className="c4"
          size="small"
        >
          test
        </h2>
      </header>
      <div>
        test description
      </div>
      <footer
        className="c5"
      >
        <span
          className=""
        >
          Stars: 
          1
        </span>
        <span
          className=""
        >
          Last update: 
          01
          /
          14
          /
          2019
        </span>
      </footer>
    </article>
  </main>
</div>
`;
